threedi_models_and_simulations/widgets/log_in.py

```diff
@@ -43,6 +51,143 @@ def wrapper(self):
     return wrapper
 
 
+class AuthorizationCodeInterceptor(QObject):
+    """Authorization code interceptor class."""
+
+    code_intercepted = pyqtSignal(str)
+    code_interception_failed = pyqtSignal(Exception)
+
+    def __init__(self, authorization_handler, state=None):
+        super().__init__()
+        self.authorization_handler = authorization_handler
+        self.state = state
+        self.tcp_server = None
+
+    def start_listening(self, host=QHostAddress.LocalHost, port=1234):
+        """Start listening for an incoming connections."""
+        self.tcp_server = QTcpServer(self)
+        if not self.tcp_server.listen(host, port):
+            raise RuntimeError(f"Can't listen port: '{port}'!")
```

Comment:
I recommend picking a rarely used port + fallback port. We can configure 2 redirect uris, no problem. Pick two random numbers in the 49152-65535 range.